dsl_parser/functions.py

```diff
@@ -35,6 +35,63 @@
 TEMPLATE_FUNCTIONS = {}
 
 
+class RecursionLimit(object):
+    def __init__(self, limit, args_to_str_func=None):
+        """Initializes the recursion limit context manager. This context
+        manager limits the recursion only for this specific function, unlike
+        the sys.setrecursionlimit(limit) function, which sets the call stack
+        depth.
+
+        :param limit: recursion limit (inclusive).
+        :param args_to_str_func: function that returns a string containing
+            information about the last arguments used. It's signature should
```

Comment:
<pedant> Its signature should be... 
Possessive it does not require apostrophes.
</pedant>

Reply:
It removed by pedant tags!

Reply:
changed the docstring, however note that this wasn't added in this pr